tests/nn/data_parallel/test_fsdp_context.py

```diff
@@ -0,0 +1,52 @@
+import torch.nn as nn
```

Comment:
is there a test here for recursive_wrap? It would be good to have a simple one for now, can extend as we see more use cases. it is tricky since we perhaps don't want to wrap inner module where the outer module reaches into the inner one during forward().

Reply:
I added a very rudimentary test! I think currently I see `auto_wrap` as providing very little flexibility (and a few assumptions that a sequential module has, i.e guarantees we're just calling the forward function) and `wrap` doing the most flexible of things.

The great thing is we can interchange/swap them as we see fit, i.e `auto_wrap` the modules we know are supported and do not do any child module magic other than call `forward`, and just use `wrap` for the cases where we'd not want to modify the child modules.